geopandas/tests/test_geodataframe.py

```diff
@@ -772,6 +772,27 @@ def test_dataframe_to_geodataframe(self):
         with pytest.raises(ValueError):
             df.set_geometry("location", inplace=True)
 
+    def test_dataframe_not_manipulated(self):
+        df = pd.DataFrame(
+            {
+                "A": range(len(self.df)),
+                "latitude": self.df.geometry.centroid.y,
+                "longitude": self.df.geometry.centroid.x,
+            },
+            index=self.df.index,
+        )
+        df_copy = df.copy()
+        gf = GeoDataFrame(
+            df,
+            geometry=points_from_xy(df["longitude"], df["latitude"]),
+            crs=self.df.crs,
+        )
+        assert isinstance(df, pd.DataFrame)
+        assert "geometry" not in df
+        pd.testing.assert_frame_equal(df, df_copy)
+        assert isinstance(gf, GeoDataFrame)
+        assert hasattr(gf, "geometry")
+
```

Comment:
Perhaps overkill, but I'm wondering if it also makes sense to explicitly check that  mutating the columns in gf doesn't update df:
```suggestion
gf.loc[0, "A" ] =7
pd.testing.assert_frame_equal(df, df_copy)
gf["A"] =3
pd.testing.assert_frame_equal(df, df_copy)
```

Reply:
added in in https://github.com/geopandas/geopandas/pull/2667/commits/22c4a91923f89fccf98c70ad64ceb603284fa65d